### DISCUSSION TITLE
Why rc.d?

### DISCUSSION THREAD

**Discussion by fagci** [marlonrichert/zsh-launchpad#4] [Q&A] [answered]
Explicit sourcing of scripts best __for me__ in reasons:
1. vim goto file works good
2. faster init time
3. disable sourcing of files faster by comment it

Ofc, add another file into rc.d directory will be faster. But I think, that any multiple additions will decrease performance.

So, why rc.d? =)

**Comment by marlonrichert** [comment, marked as answer]
The overhead of sourcing a file is negligible in `zsh`. On my machine, there's a delay of only 0.000012 to 0.000025 seconds between the evaluation of the `source` or `.` statement and the evaluation of the first statement in the sourced file. The total overhead of all `.` calls in the `.zshrc` file here combined is only 0.000133 seconds and the evaluation of `$ZDOTDIR/rc.d/<->-*.zsh(n)` takes just 0.000037 seconds. I think we can live with that. 🙂 

As for your question "Why `rc.d`?", I have two answers:
* `git` is better at detecting file renames than at keeping track of blocks of code being moved inside the same file, especially if you've made other modifications, too.
* When reading the config, it's much easier to navigate between multiple small files than between multiple sections of one very large file.
* Numbering the files makes it immediately apparent in which order they are sourced and means they will also be listed in the correct order.
* It lets you add or remove files in your config without having to edit any other files.

**Reply by fagci** [reply]
Overhead so low, it's ok.
I think, rc.d concept and direct sourcing can be live as thing of taste =)

**Reply by fagci** [reply]
Also, to speedup zsh (not only startup time), I just comment out some sourced scripts and get result faster, compared with moving or renaming files.

**Reply by fagci** [reply]
> * When reading the config, it's much easier to navigate between multiple small files than between multiple sections of one very large file.

So, you are about large file initially, that is bad choice for launchpad in perspective.

With multiple files you get many "blank" files to fill =)

**Reply by marlonrichert** [reply]
You should, of course, organize your dotfiles in whatever way makes sense to you. I organized the files in this way based on the content I want to present; if I had different content, I would’ve organized it differently. Remember, as the Readme states, this is supposed to be used as a _template_ for your own dotfiles. You can and should customize it to your own needs. 

> Also, to speedup zsh (not only startup time), I just comment out some sourced scripts and get result faster, compared with moving or renaming files.

If you don’t need it, just delete it. When in doubt, throw it out.

**Comment by fagci** [comment]
> But I think, that any multiple additions will decrease performance.

Here I mean performance degrade with tons of code inside, not globbing itself. 
To make things tight, do no place tons of scripts inside rc.d. Just source what you need. And only this scripts will be loaded.

So, direct sourcing here will be better in terms of explicity. This is my philosophy. =)

**Reply by marlonrichert** [reply]
I am not sourcing anything unnecessary in these dotfiles. I’m only sourcing what I want to source. What is inside `rc.d` is not random; it’s what I’ve deliberately put there. Remember, as the Readme states, this is supposed to be used as a _template_ for your own dotfiles. You can and should customize it to your own needs.

**Reply by fagci** [reply]
Because of that your answer was accepted
